File: app/src/main/java/pe/andy/bookholic/util/Str.java
package pe.andy.bookholic.util;

import android.text.TextUtils;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;

public class Str {

    public static String def(String text) {
        return (text != null) ? text : "";
    }

    public static Queue<String> splitToQ(final String str, final String separatorRegex) {

        return new LinkedList<>(
            Arrays.asList(
                TextUtils.split( (str != null) ? str : "",
                        (separatorRegex != null) ? separatorRegex : "")
            )
        );
    }

    public static int extractInt(final String str) {
        if (str == null)
            return 0;

        try {
            return Integer.parseInt(str.replaceAll("\\D*", ""));
        }
        catch (Exception e){
            return 0;
        }
    }

}

File: app/src/main/java/pe/andy/bookholic/util/Str.kt
package pe.andy.bookholic.util


class Str {
    companion object {
        @JvmStatic
        fun def(text: String?): String {
            return text ?: ""
        }

        private val nonDigitPattern: Regex = Regex("\\D*")

        @JvmStatic
        fun extractInt(str: String = "0"): Int {
            return try {
                Integer.parseInt(str.replace(nonDigitPattern, ""))
            } catch (e: Exception) {
                0
            }
        }
    }
}